lib/runtime/wasmer/exports.go

```diff
@@ -86,6 +86,60 @@ func (in *Instance) GrandpaAuthorities() ([]types.Authority, error) {
 	return types.GrandpaAuthoritiesRawToAuthorities(gar)
 }
 
+// BabeGenerateKeyOwnershipProof returns babe key ownership proof from runtime.
+func (in *Instance) BabeGenerateKeyOwnershipProof(slot uint64, authorityID [32]byte) (
+	types.OpaqueKeyOwnershipProof, error) {
+
+	combinedArg := []byte{}
+	encodedSetID, err := scale.Marshal(slot)
+	if err != nil {
+		return nil, fmt.Errorf("encoding set id: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedSetID...)
+
+	encodedAuthorityID, err := scale.Marshal(authorityID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding authority id: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedAuthorityID...)
```

Comment:
You might want to use the `scale.Encoder` to directly write to the final buffer, and also pre-allocate the max size for that buffer which can be known in advance:

```go
const maxBufferLength = 8 + 33 // scale encoded slot uint64 + scale encoded array of 32 bytes
buffer := bytes.NewBuffer(make([]byte, 0, maxBufferLength))
encoder := scale.NewEncoder(buffer)
err = encoder.Encode(slot)
// ...
err = encoder.Encode(authorityID)
// ...
ret, err := in.Exec(runtime.BabeAPIGenerateKeyOwnershipProof, buffer.Bytes())
```

Reply:
Curious to see what method gets chosen here, will follow suit with what @kishansagathiya does in my grandpa version of this PR

Reply:
You can also put it into a helper struct and just encode that instead of appending the encodings.

Reply:
> You can also put it into a helper struct and just encode that instead of appending the encodings.

Have gone with this one, seem more appropriate